text/0000-kebab-actions.md

```diff
@@ -0,0 +1,208 @@
+
+If any non-function is passed to a kebab, such as in
+`on-click="alert('blah');"` an assertion should fail. Only functions are
+permitted.
+
+#### Handling a Kebab Action
+
+Kebab actions will be invoked with the raw DOM event (not the jQuery event)
+as an argument. For example:
+
+```js
+// app/components/my-button.js
+export Ember.Component.extend({
+  actions: {
+    save(event) {
+      console.log('saved with click on ', event.target);
+      this.get('model').save();
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-button.hbs }}
+<button on-click={{action 'save'}}>Save</button>
+```
+
+Actions curry arguments, thus the event may not always be the first argument. For
+example:
+
+```js
+// app/components/my-input.js
+export Ember.Component.extend({
+  actions: {
+    log(prefix, event) {
+      console.log(prefix, Ember.$(event.target).val());
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-input.hbs }}
+<input on-change={{action 'log' 'value is:'}} />
+```
+
+Additionally, the `value` argument to `action` is useful when passing the
+event.
+
+
+```js
+// app/components/my-input.js
+export Ember.Component.extend({
+  actions: {
+    log(value) {
+      console.log(target);
+    }
+  }
+});
+```
+
+```hbs
+{{! app/templates/components/my-input.hbs }}
+<input on-change={{action 'log' value='target.value'}} />
+```
+
+Passing the event is useful for interacting with the originating DOM node,
+but additionally important for allowing event propagation to be controlled
+(cancelled).
+
+#### Event Management
+
+Kebab actions are lazy. If a kebab action of `on-flummux` is used, then Ember
+should listen for the event of `flummux` on the root element and dispatch
+that action when it fires.
+
+If an event from the [list of events Ember listens for](http://emberjs.com/api/classes/Ember.View.html#toc_event-names)
+is used, then kebabs can use the already existing listeners.
+
+# Drawbacks
+
+Obviously, if this is an RFC, there cannot be drawbacks.
+
+j/k.
+
+Kebab actions, as described here, use a dasherized multi-word format. For
+example `on-mouse-over`. This corresponds to how Ember already camelizes
+event listeners (for example `mouseOver` is the method for a component
+listener), but does *not* correspond to the native browser APIs which would
+imply `on-mouseover`. As what form of camelization, dasherization, or single-word-ness
+to use can be extremely confusing to a newcomer, I am inclined to suggest
+matching the native browser instead of following Ember's convention of splitting
+the words.
+
+# Alternatives
+
+One immediate alternative making the rounds today is using event listeners
+directly. For example:
+
+```hbs
+<button onclick={{action 'save'}}></button>
+```
+
+The above passes a function to the `onclick` property of the `<button>` DOM
+node. The upside of recommending this usage is the directness of it: You can
+demonstrate what is going on trivially. However the downside is that many
```

Comment:
This weight is actually pretty relatively negligible, in the past this was more costly. Also, its worth noting the its slow to add eventHandlers meme, is largely related to the jQuery selector parsing and DOM walking required to find the elements to attach listeners too using `addEventListener`.

Even delegation has other values, but un-related to this. The ability to say any time any X gets clicked, please invoke this function. But that is already rendered irrelevant since the user is declaring even handlers explicitly.